### DISCUSSION TITLE
Add profile backup service to SPT

### DISCUSSION THREAD

**Discussion by DrakiaXYZ** [sp-tarkov/server#982] [Ideas]
Currently users need mods (Which aren't often updated right away) to back up their profile and avoid painful experiences with misconfigured mods and profile corruption. We could add a built in simple profile backup service, where we only backup the profile on login (Game start), it wouldn't be as flexible as the previous profile backup mods, but would help users avoid having to create new profiles if they make a mistake with a mod's configuration

Proposed backup back: `profiles/backups/{profileId}/{date}/{profileId}.json`

We would want to keep the JSON file named the same as the actual profile JSON to avoid users needing to rename anything to restore a backup. Keep everything simple

We would keep the previous X amount of backups per profile, whether we make X configurable or just pick a number is open for discussion

**Comment by chompDev** [comment]
I feel its important to preserve the profiles name to help when transferring it from backup to an active location.
Another issue is when do we take the backup?, i feel we need to take a snapshot before mods have loaded (and made possible changes)

What would be a good period between backups? hourly? every server start? if its every server start, how many backups are preserved before they're purged?

I would like to hear from people helping others with profile issues

**Comment by refringe** [comment]
I know for a fact that the insurance system will check for and modify all profile's insurance packages if they're due... It would probably be best to back them *all* up before they're touched on server start.

**Comment by refringe** [comment]
I've created a draft PR for the base functionally here: https://github.com/sp-tarkov/server/pull/983
